sklearn/neighbors/tests/test_sample_weight.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+import pytest
+import scipy.stats
+
+from sklearn import neighbors
+from sklearn.utils._testing import assert_array_almost_equal
+
+
+@pytest.mark.parametrize("weights", ["uniform", "distance", scipy.stats.norm.pdf])
+def test_KNeighborsRegressor_sample_weight_none_is_same_as_uniform(
+    weights,
+    n_samples=40,
+    n_features=5,
+    n_test_pts=10,
+    n_neighbors=3,
+    random_state=0,
+):
+    rng = np.random.RandomState(random_state)
+    X = 2 * rng.rand(n_samples, n_features) - 1
+    y = np.sqrt((X ** 2).sum(1))
+    y /= y.max()
+
+    epsilon = 1e-5 * (2 * rng.rand(1, n_features) - 1)
+
+    def _predict(sample_weight):
+        knn = neighbors.KNeighborsRegressor(n_neighbors=n_neighbors, weights=weights)
+        knn.fit(X, y, sample_weight)
+        return knn.predict(X[:n_test_pts] + epsilon)
+
+    y_pred_sample_weight_none = _predict(None)
+    y_pred_sample_weight_np_ones = _predict(np.ones(X.shape[0]))
+    y_pred_sample_weight_np_twos = _predict(2 * np.ones(X.shape[0]))
+
+    assert_array_almost_equal(y_pred_sample_weight_none, y_pred_sample_weight_np_ones)
+    assert_array_almost_equal(y_pred_sample_weight_none, y_pred_sample_weight_np_twos)
+
+
+def test_KNeighborsRegressor_sample_weight():
+    num_samples = 2
+    X = np.array(range(num_samples)).reshape(-1, 1)
+    y = 10 + np.array(range(num_samples))
+
+    num_test = 10
+    X_test = np.linspace(0, num_samples - 1, num=num_test).reshape(-1, 1)
+
+    for alpha in (0, 0.25, 0.5, 0.75, 1):
```

Comment:
Instead of an internal for loop, it is better to parametrize the test

Reply:
ok